File: 백준/Silver/1002. 터렛/터렛.py
import sys
import math
input = sys.stdin.readline

T = int(input())
for _ in range(T):
    x1, y1, r1, x2, y2, r2 = map(int, input().split())
    distance = math.sqrt((x1 - x2) ** 2 + (y1 - y2) ** 2)
    if distance == 0 and r1 == r2: # 두 원이 일치할 경우
        print(-1)
    elif abs(r1-r2) == distance or r1+r2 == distance:
        print(1) # 두 원이 내접이나 외접할 때
    elif abs(r1-r2) < distance < r1+r2:
        print(2) # 두 원이 서로 다른 두 점에서 만날 경우
    else:
        print(0)

File: 백준/Silver/1002. 터렛/터렛.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

const T = Number(input[0]);
const results = [];

for (let i = 1; i <= T; i++) {
  const [x1, y1, r1, x2, y2, r2] = input[i].split(" ").map(Number);
  const distanceSquared = (x1 - x2) ** 2 + (y1 - y2) ** 2; // 거리의 제곱
  const rSumSquared = (r1 + r2) ** 2; // 두 반지름 합의 제곱
  const rDiffSquared = (r1 - r2) ** 2; // 두 반지름 차이의 제곱

  if (distanceSquared === 0 && r1 === r2) {
    results.push("-1");
  } else if (
    distanceSquared === rSumSquared ||
    distanceSquared === rDiffSquared
  ) {
    results.push("1");
  } else if (rDiffSquared < distanceSquared && distanceSquared < rSumSquared) {
    results.push("2");
  } else {
    results.push("0");
  }
}

console.log(results.join("\n"));
